src/identifierToProjectId.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  FirebaseProjectMetadata,
+  getFirebaseProject,
+  listFirebaseProjects,
+} from "./management/projects";
+
+async function getProjectIdFromDatabaseInstance(instance: string): Promise<string | undefined> {
+  const projects = await listFirebaseProjects();
+
+  const matchedProject = projects.find((p) => {
+    if (!p.resources) {
+      return false;
+    }
+    return p.resources.realtimeDatabaseInstance === instance;
+  });
+
+  return matchedProject && matchedProject.projectId;
+}
+
+/**
+ * Resolve to the project ID of the Firebase project whose project ID or realtime database instance
+ * equals to the input `id`.
+ */
+export default async function(id: string): Promise<string | undefined> {
```

Comment:
Please compare this with the old version. I tried to make sure that it behaves the same as before.